samples/app-complete-sample/nodejs-v4/Readme.md

```diff
@@ -0,0 +1,102 @@
+---
+page_type: sample
+products:
+- office-teams
+- office-365
+languages:
+- nodejs
+- javascript
+extensions:
+  contentType: samples
+  technologies:
+  - Tabs
+  createdDate: 5/8/2017 9:41:53 AM
+---
+
+# Microsoft Teams Bot in Node.js
+
+Sample that shows how to build a Microsoft Teams bot in Node.js.
+
+## Features of this sample
+
+![](images/dialog.png)
+
+![](images/static-tab.png)
+
+![](images/messaging-extension.png)
+
+
+## Prerequisites
+
+- [NodeJS](https://nodejs.org/en/)
+- [ngrok](https://ngrok.com/) or equivalent tunnelling solution
+
+## To try this sample
+
+1) Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+2) In a terminal, navigate to `samples/auto-complete-sample/nodejs-v4
+`
+
+3) Install modules
+
+    ```bash
+    npm install
+    ```
+
+4) Run ngrok - point to port 3979
+
+    ```bash
+    ngrok http -host-header=rewrite 3979
+    ```
+
+5). Register a new bot (or update an existing one) with Bot Framework by using the full endpoint as the bot's "Messaging endpoint". Make sure to add Teams as a channel for the bot. Bot registration is here (open in a new browser tab): https://dev.botframework.com/bots
+
+    > **NOTE**: When you create your bot you will create an App ID and App password - make sure you keep these for later.
+
+6). Once you have saved your bot and gotten the confirmation that it is created, navigate back to your  project. Open the ".env" file. There, copy/paste your App ID and App password from the step above.
+    ```
+    MicrosoftAppId=
```

Comment:
Lets's add connection name setting as well and how to get app id and secret while creating facebook connection.